tests/indexes/test_api.py

```diff
@@ -451,3 +455,68 @@ async def test_delete_index(spawn_job_client, error):
         assert 204 == response.status
         async for doc in history.find({"index.id": index_id}):
             assert doc["index"]["id"] == doc["index"]["version"] == "unbuilt"
+
+
+@pytest.mark.parametrize("error", [None, "400_exists", "400_name", "404", "422"])
+async def test_upload(error, tmpdir, spawn_client, snapshot, resp_is, pg_session):
+    client = await spawn_client(authorize=True)
+    path = Path(sys.path[0]) / "tests" / "test_files" / "index" / "reference.1.bt2"
+
+    files = {
+        "file": open(path, "rb")
+    }
+
+    client.app["settings"]["data_path"] = str(tmpdir)
+
+    index = {
+        "_id": "foo",
+        "reference": {
+            "id": "bar"
+        },
+        "user": {
+            "id": "test"
+        }
+    }
+
+    if error == "400_exists":
+        index["files"] = [1]
+
+    await client.db.indexes.insert_one(index)
+
+    url = "/api/indexes/foo/files"
+
+    if error == "422":
+        url += "?type=fasta"
+    elif error == "400_name":
+        url += "?name=reference.bt2"
+    else:
+        url += "?name=reference.1.bt2"
+
+    resp = await client.post_form(url, data=files)
+
+    if error == "400_name":
+        assert await resp_is.bad_request(resp, "Unsupported index file name")
+        return
+
+    if error == "400_exists":
```

Comment:
Should be `409`.

https://stackoverflow.com/questions/9269040/which-http-response-code-for-this-email-is-already-registered/9270432#:~:text=409%20%3D%3E%20Conflict%20That%20mean.,exists%20in%20the%20parent%20location.